legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/stores/StoreExecutableManager.java

```diff
@@ -34,6 +36,8 @@
 {
     INSTANCE;
     private final ConcurrentHashMap<String, List<StoreExecutable>> requestExecutableMap = new ConcurrentHashMap<>();
+
+    private final ConcurrentHashMap<String, List<String>> sessionIDToProvidedID = new ConcurrentHashMap<>();
```

Comment:
The way things are been use, seems better to leverage a Set rather than a List. 

This will help on the remove statements, while also avoiding the need of checking if contains, as duplicates will have no effect.  

Thoughts?